### ISSUE TITLE
Mixed subscription issue

### ISSUE THREAD

**Issue by akiomik** [penpenpng/rx-nostr#1]
Thanks for providing this cool library.

I was trying `rx-nostr@v0.0.3` with the code in `app/main.ts`, but when I do two or more subscriptions at the same time, the events get mixed up.

```node
import { debounceTime } from 'rxjs';
import { Relays, BackwardReq, MonoFilterAccumulater, uniq } from 'rx-nostr';
import { webcrypto } from 'crypto';
import WebSocket from 'ws';

global.WebSocket = WebSocket;
global.crypto = webcrypto;

const relays = new Relays([
  "wss://relay-jp.nostr.wirednet.jp",
  "wss://nostr-relay.nokotaro.com",
]);

const acc0 = new MonoFilterAccumulater({ kinds: [0], limit: 5 });
const req0 = BackwardReq.from(acc0, debounceTime(500));

const acc1 = new MonoFilterAccumulater({ kinds: [1], limit: 5 });
const req1 = BackwardReq.from(acc1, debounceTime(500));

relays.observeReq(req0).pipe(uniq()).subscribe((e) => console.log('req0', e.message[2].kind));
relays.observeReq(req1).pipe(uniq()).subscribe((e) => console.log('req1', e.message[2].kind));
```

```bash
# output
# req0 expects to get only kind0 and req1 expects to get only kind1
req0 1
req1 1
req0 1
req1 1
req0 1
req1 1
req0 1
req1 1
req0 1
req1 1
req0 0
req1 0
req0 0
req1 0
req0 0
req1 0
req0 0
req1 0
req0 0
req1 0
```

Am I using it incorrectly?
As far as I've tried, it reproduced in both node and browser.

Full code is here: https://github.com/akiomik/rx-nostr-playground/blob/859e6be14512023b788eddf650070c2e24683736/main.js

**Comment by penpenpng** [comment]
No doubt this is a bug. However, I am in the process of refactoring, so please let this issue "won't fix" for now.

If this is no longer the case in newer versions, this issue will be closed.

**Comment by penpenpng** [comment]
fixed in 0.1.0
